cpp/src/arrow/compute/exec/asof_join_benchmark.cc

```diff
@@ -76,7 +75,7 @@ static void TableJoinOverhead(benchmark::State& state,
   for (auto _ : state) {
     state.PauseTiming();
     ASSERT_OK_AND_ASSIGN(std::shared_ptr<arrow::compute::ExecPlan> plan,
-                         ExecPlan::Make(&ctx));
+                         ExecPlan::Make());
```

Comment:
This line and similar ones seem like an unintended change.

Reply:
I think these are leftover from your brief merge on top of the PR that adds the proper single-threaded execution mode.  In that PR `ExecPlan::Make` no longer takes an exec context since the exec context has a thread pool and we don't want to actually provide or start using the thread pool until we start producing.  It's somewhat harmless here since `ctx` is identical to the default exec context.  However, it is not harmless in `substrait/util.cc` because the context there is actually using the CPU thread pool.